search/binary_Search.dart

```diff
@@ -1,32 +1,32 @@
-int binary_search(List a, int l, int r, int x) {
+int binary_search(List num, int l, int r, int search) {
   if (r >= l) {
     int middle = (l + (r - l) / 2).toInt();
 
     //If the element is present at middle
-    if (a[middle] == x) {
+    if (num[middle] == search) {
       return middle;
     }
 
     //If the element is smaller than middle
-    if (a[middle] > x) {
-      return binary_search(a, l, middle - 1, x);
+    if (num[middle] > search) {
+      return binary_search(num, l, middle - 1, search);
     }
 
-    return binary_search(a, middle + 1, r, x);
+    return binary_search(num, middle + 1, r, search);
   }
   return -1;
 }
 
 void main() {
   List list = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89];
-  int x = 55;
+  int search = 55;
   int n = list.length;
-  int index = binary_search(list, 0, n - 1, x);
+  int index = binary_search(list, 0, n - 1, search);
   print('list:');
   print(list);
   if (index != -1) {
-    print('$x found at positions: $index');
+    print('$search found at positions: $index');
   } else {
-    print('$x Not found');
+    print('$search Not found');
   }
```

Comment:
Since, We are improiving this file. I would recommend you add some test to this file as well.